packages/gatsby-link/src/index.js

```diff
@@ -4,6 +4,11 @@ import React from "react"
 import { Link, Location } from "@reach/router"
 import { parsePath } from "gatsby"
 
+// Temp hack while awaiting https://github.com/reach/router/issues/119
+if (typeof window !== `undefined`) {
+  window.__navigatingToLink = false
+}
+
```

Comment:
I would move this to https://github.com/gatsbyjs/gatsby/blob/5f09cc377d06e51df5b39abc2fb5300853907c6b/packages/gatsby/cache-dir/navigation.js#L113